agent/lib/kontena/workers/weave_worker.rb

```diff
@@ -95,5 +69,51 @@ def weave_attach(container)
       error exc.backtrace.join("\n")
     end
 
+    # @param [String] container_id
+    # @param [String] overlay_cidr
+    def attach_overlay(container_id, overlay_cidr)
+      Actor[:network_adapter].exec(['--local', 'attach', overlay_cidr, '--rewrite-hosts', container.id])
+    end
+
+    # @param [Docker::Container]
+    def register_container_dns(container)
+      grid_name = container.labels['io.kontena.grid.name']
+      service_name = container.labels['io.kontena.service.name']
+      instance_number = container.labels['io.kontena.service.instance_number']
+      domain_name = container.config['Domainname'] || "#{grid_name}.kontena.local"
+      if container.default_stack?
+        hostname = container.labels['io.kontena.container.name']
+        dns_names = default_stack_dns_names(hostname, service_name, domain_name)
+        dns_names = dns_names + stack_dns_names(hostname, service_name, domain_name)
+
+      else
+        hostname = container.config['Hostname']
+        dns_names = stack_dns_names(hostname, service_name, domain_name)
+      end
+      dns_names.each do |name|
```

Comment:
Yes, makes sense in a way.. but is it clear that if I call `container.each_dns_name` that it will return dns names that container _should_ have, but not necessarily have them before they are set to weave dns? 

Anyways not all containers have dns names so it might be a bad idea to bundle them to the Container class.